pancancer_utilities/tcga_utilities.py

```diff
@@ -0,0 +1,180 @@
+"""
+Functions for preprocessing TCGA expression data and mutation status labels.
+
+Most functions are adapted from:
+https://github.com/greenelab/BioBombe/blob/master/9.tcga-classify/scripts/tcga_util.py
+"""
+import os
+
+import pandas as pd
+from sklearn.preprocessing import StandardScaler
+
+def process_y_matrix(
+    y_mutation,
+    y_copy,
+    include_copy,
+    gene,
+    sample_freeze,
+    mutation_burden,
+    filter_count,
+    filter_prop,
+    output_directory,
+    hyper_filter=5,
+):
+    """
+    Combine copy number and mutation data and filter cancer-types to build y matrix
+
+    Arguments
+    ---------
+    y_mutation: Pandas DataFrame of mutation status
+    y_copy: Pandas DataFrame of copy number status
+    include_copy: boolean if the copy number data should be included in status calc
+    gene: string indicating gene of interest (used for writing proportion file)
+    sample_freeze: pandas dataframe storing which samples to use
+    mutation_burden: pandas dataframe storing log10 mutation counts
+    filter_count: the number of positives or negatives required per cancer-type
+    filter_prop: the proportion of positives or negatives required per cancer-type
+    output_directory: the name of the directory to store the gene summary
+    hyper_filter: the number of std dev above log10 mutation burden to filter
+
+    Returns
+    -------
+    Write file of cancer-type filtering to disk and output a processed y vector
+    """
+    if include_copy:
+        y_df = y_copy + y_mutation
+    else:
+        y_df = y_mutation
+
+    y_df.loc[y_df > 1] = 1
+    y_df = pd.DataFrame(y_df)
+    y_df.columns = ["status"]
+
+    y_df = (
```

Comment:
Might be a little lost here...so `y_df` is a df with a single `status` column? Does that mean that mutation burden and copy status values are stacked on top of each other? And I assume the rows are sample ids. If there is only one column doesn't that means that you'll have duplicate sample ids since I assume samples will have mutation and copy status info associated?

Reply:
> so y_df is a df with a single status column? Does that mean that mutation burden and copy status values are stacked on top of each other?

Yep, exactly. We're defining a "mutation" to be any deleterious point mutation *or* copy number change in the given gene. So the line at the top of the function `y_df = y_copy + y_mutation` is taking the sum of CNVs and point mutations in the gene.

> doesn't that means that you'll have duplicate sample ids since I assume samples will have mutation and copy status info associated?

I'll admit that I just copied this line from Greg's code, but I think taking the sum of dataframes that have the same indexes (sample IDs in this case) should just take the sum of each of the entries. The list of sample IDs should be the same, then, after taking the sum. I'll check this by running some simple examples in a notebook.